pkg/limitador/k8s_objects.go

```diff
@@ -52,7 +53,7 @@ func Service(limitador *limitadorv1alpha1.Limitador) *v1.Service {
 	}
 }
 
-func Deployment(limitador *limitadorv1alpha1.Limitador, deploymentOptions DeploymentOptions) *appsv1.Deployment {
+func Deployment(limitador *limitadorv1alpha1.Limitador, deploymentOptions DeploymentOptions, logger logr.Logger) *appsv1.Deployment {
```

Comment:
@eguzki @KevFan are you fine with passing the logger here, it requires additional setup in the test. the other options is to setup the labels and either pass those seperately or add them to the deploymentOptions and access them there inside this function.

Reply:
up to you :) matter of taste